packages/preview-middleware/src/base/flp.ts

```diff
@@ -481,23 +483,64 @@ export class FlpSandbox {
      * @param manifest manifest of the additional target app
      * @param app configuration for the preview
      */
-    addApp(manifest: Manifest, app: App) {
+    async addApp(manifest: Manifest, app: App) {
         const id = manifest['sap.app'].id;
         app.intent ??= {
             object: id.replace(/\./g, ''),
             action: 'preview'
         };
         this.templateConfig.ui5.resources[id] = app.target;
         this.templateConfig.apps[`${app.intent?.object}-${app.intent?.action}`] = {
-            title: manifest['sap.app'].title ?? id,
-            description: manifest['sap.app'].description ?? '',
+            title: (await this.getI18nTextFromProperty(app.local, manifest['sap.app'].title)) ?? manifest['sap.app'].id,
+            description: (await this.getI18nTextFromProperty(app.local, manifest['sap.app'].description)) ?? '',
             additionalInformation: `SAPUI5.Component=${app.componentId ?? id}`,
             applicationType: 'URL',
             url: app.target,
             applicationDependencies: { manifest: true }
         };
     }
 
+    /**
+     * Get the i18n properties bundle.
+     *
+     * @param projectRoot absolute path to the project root
+     * @returns i18n properties bundle
+     * @private
+     */
+    private async getPropertiesI18nBundle(projectRoot: string) {
+        //todo: consider relative path from manifest['sap.app'].i18n
+        //todo: consider locale
+        const i18nFilePath = join(projectRoot, 'webapp', 'i18n', 'i18n.properties');
```

Comment:
this would only work for projects that have not set a custom `webapp` path. Module `@sap-ux/project-provider` provides convenient methods for retrieving the i18n bundles and also consider custom webapp paths, see https://github.com/SAP/open-ux-tools/blob/edc3a166650d2d3e1981372cd07b3ba1811b40d8/packages/project-access/src/project/access.ts#L143

Otherwise `@sap-ux/project-provider` has also a function to get custom webapp paths:
https://github.com/SAP/open-ux-tools/blob/edc3a166650d2d3e1981372cd07b3ba1811b40d8/packages/project-access/src/project/ui5-config.ts#L14-L26

Reply:
Hi @Klaus-Keller and thanks for the review

`@sap-ux/project-provider` even deals with all variants of [JSONSchemaForSAPAPPNamespace.i18n](https://github.com/SAP/ui5-manifest/blob/main/types/manifest.d.ts#L803) 👍🏻 

I'll adjust the code.

Are there any plans to support loading the `locale` version of the `i18n.properties`? Then I could get rid of the todo here as well 😸